### DISCUSSION TITLE
Enable custom ratio for split screens instead of 50/50

### DISCUSSION THREAD

**Discussion by ssbarnea** [rxhanson/Rectangle#707] [Ideas]
One major problem with 4K and 5K displays is that both 50/50 and 66/33 hardcoded rations may prove poor.

I do have a custom hammerspoon script that allows me to obtain a 60/40 ratio, something that seems to be the sweet spot for having the IDE on wide area while still having a browser or email client on the narrow side. Using a 33% approach does produce a too narrow to be practical. Still, I would prefer to be able to use rectangle or its newer brother as they provide other features for which I did not had time to code the script, like moving between multiple displays.

This is why I think it would be preferable to be able to tune the split ratio, with a default of 50/50 for backwards compatibility.

**Comment by rxhanson** [comment]
This boils down to the necessity of a custom shortcut builder, as matching every desirable window size would end up being unwieldy for Rectangle's simple UI. My take on a custom shortcut builder is built into my paid superset of Rectangle, Hookshot (https://hookshot.app). Here's a post I wrote on using the custom shortcut builder and more details on why I didn't put it in Rectangle: https://medium.com/ryan-hanson/custom-window-dimension-shortcuts-in-macos-e7df347dfd24

**Reply by ssbarnea** [reply]
I read the article but I was not able to make it fully work. That is because the tool does not remember last sequence position and restarts. The setup below was closest to desired behavior but still fails short on several cases. I was looking to use Cmd and arrow to switch between monitors on left sign. If I setup a display for the root entry it will not work well when you are already on the right display, so I had to leave it unassigned. I guess it would be possible to obtain a desired behavior if we could use a special display value for switching to next display, like a `-1.

![](https://sbarnea.com/ss/Screen-Shot-2022-01-07-10-22-36.87.png)

I also considered trying to implement a setup of 4 columns like `[ a, b ] [ c, d ]` and use arrow based shortcuts to move between columns. The problem is that due to lack of knowledge regarding current window position it will always start cycling from root, so it would not be possible to move a window to the left of right of its current position.

I wonder if it would be possible to remember the current sequence position for each window so next time a shortcut is called you would be able to jump to the right position instead of start the movements from zero. In fact it may not even need to remember the position in sequence if it can check if current window position is matching any existing entry in sequence and assume that's where it should start from.

To explain it on my case Cmd+RightArrow has 3 entries in its sequence. When I would press the shortcut, it would first check if current window is matching any of existing 3 layouts. If one is found it will assume it needs to setup the next layout. If none is found it will just pick the first one (root), like now.
